contracts/DutchExchange.sol

```diff
@@ -764,166 +720,200 @@ contract DutchExchange {
+            uint ETHUSDPrice = PriceOracleInterface(ETHUSDOracle).getUSDETHPrice();
+            fraction memory price = priceOracle(token);
 
             // Convert fee to ETH, then USD
-            uint feeInETH = fee * tokenPriceNum / tokenPriceDen;
-            uint feeInUSD = feeInETH * PriceOracleInterface(ETHUSDOracle).getUSDETHPrice();
-            uint amountOfWIZBurned = Math.min(Token(OWL).allowance(msg.sender, this), feeInUSD / 2);
+            uint feeInETH = fee * price.num / price.den;
+            uint feeInUSD = feeInETH * ETHUSDPrice;
+            uint amountOfOWLBurned = Math.min(Token(OWL).allowance(msg.sender, this), feeInUSD / 2);
 
             //burning OWL tokens with delegatecall is risky, because this allows OWL token to modify the storage of this contract.
             // OWL.delegatecall(bytes4(sha3("burnOWL(uint256)")), amount);
 
             // Adjust fee
-            fee = feeInETH * amountOfWIZBurned / feeInUSD;
+            if (feeInUSD != 0) {
+                fee -= amountOfOWLBurned * fee / feeInUSD;
+            }
         }
     }
 
-
-    /// @dev Gives best estimate for market price of a token in ETH of any price oracle on the Ethereum network
-    /// @param buyToken - address of ERC-20 token
-
-    
-    function getClosingPriceNum(
+    function scheduleNextAuction(
         address sellToken,
-        address buyToken,
-        uint auctionIndex
-    )
-        public 
-        view
-        returns (uint) 
-    {
-        return closingPrices[sellToken][buyToken][auctionIndex].num;
-    }
-
-
-    function getClosingPriceDen(
-        address sellToken,
-        address buyToken,
-        uint auctionIndex
-    )
-        public 
-        view
-        returns (uint) 
-    {
-        return closingPrices[sellToken][buyToken][auctionIndex].num;
-    } 
-
-    function getLatestAuctionIndex(
-        address sellToken,        
         address buyToken
     )
-        public 
-        view
-        returns (uint) 
+        internal
     {
-        return latestAuctionIndices[sellToken][buyToken];
+        // Check if auctions received enough sell orders
+        uint ETHUSDPrice = PriceOracleInterface(ETHUSDOracle).getUSDETHPrice();
+        uint sellVolumeNext = sellVolumesNext[sellToken][buyToken] * ETHUSDPrice;
+        uint sellVolumeNextOpp = sellVolumesNext[buyToken][sellToken] * ETHUSDPrice;
+        if (sellVolumeNext >= thresholdNewAuction || sellVolumeNextOpp >= thresholdNewAuction) {
+            // Schedule next auction
+            setAuctionStart(sellToken, buyToken, 10 minutes);
+        } 
     }
-    
+
     /// @dev Gives best estimate for market price of a token in ETH of any price oracle on the Ethereum network
     /// @param token address of ERC-20 token
     /// @return Weighted average of closing prices of opposite Token-ETH auctions, based on their sellVolume  
     function priceOracle(
         address token
+    )
+        public
```

Comment:
This function is no longer `constant`, so can't be called from JavaScript to get a meaningful result, only transaction receipt. Is this change intentional? Also it returns a `struct`, same issue as above with `getPrice` function.

Reply:
made it constant again